src/devhostagent/PortForward/ServicePortForwardConnector.cs

```diff
@@ -196,25 +197,35 @@ private async Task HookupStreamData(StreamHookData s)
         /// </remarks>
         private byte[] GetModifiedBytesForManagedIdentity(byte[] data)
         {
-            const string headerToInsert = "Metadata: true\n";
-            var stringContent = Encoding.Default.GetString(data.ToList().ToArray());
+            const string headerToInsert = "Metadata: true";
+            var stringContent = Encoding.Default.GetString(data);
+            // This log may not work as expected depending on the character on the string, adding print line inside 
+            // the for lop below to avoid wrong information while debugging
             _log.Verbose($"{loggingPrefix} Original data : {stringContent}");
             stringContent = stringContent.Replace("2017-09-01", "2018-02-01").Replace("clientid", "client_id");
             var lines = stringContent.Split(new char[] { '\n' }).ToList();
             int index = -1;
             for (int i = 0; i < lines.Count; i++)
             {
-                if (lines[i].Contains($"secret: {Common.Constants.ManagedIdentity.SecretValue}"))
+                _log.Verbose($"{loggingPrefix} Printing line : {lines[i]}");
+                if (lines[i].Contains($"secret: {Common.Constants.ManagedIdentity.SecretValue}", StringComparison.OrdinalIgnoreCase))
                 {
                     index = i;
                     break;
                 }
             }
-            lines.Insert(index + 1, headerToInsert);
+            if (index != -1) {

```

Comment:
🥷☕️ Please note: This is **interesting**, why have we changed this? Reason I ask is - previously is was `lines.Insert(index + 1, headerToInsert);` which will take care of all the cases, which means if its `-1` then place the value on the `0th` index because `-1 + 1 == 0` otherwise for any `non-0th` index use the index value. Line-207 - guarantee that first inter will be `-1` 

Why are we changing this to specific if-and-else please?

Reply:
This is an HTTP request, we are changing the raw request, if we insert Metadata:True as first line (aka index 0) it will invalidate the request since the first line should start with request type (GET, POST, etc). If you check the description of the PR I have added the logs of what it looked like when I started debugging, it changed my request from GET /metadata/identity/oauth2/token?api-versio.... to Metadata: true\n\nGET /metadata/identity/oauth2/token?api-versio... 

Basically original code had an assumption that the secret header was always present, but if it isnt we should append the metadata header at the end not at the begining. Hope this helps.